docs/network-spec/connection-manager.tex

```diff
@@ -273,7 +356,212 @@ \section{Components}
   \label{tik:components}
 \end{figure}
 
-One simple illustration of how these 3 components interact together:
+The connection manager is there to manage the underlying TCP connection
+resources. It has to provide an interface to the outbound side to enable the
+use of connections in an outbound direction. Correspondingly it must provide an
+interface to the inbound side to enable the use of connections in an inbound
+direction. Internally it must deal with connections being used in a
+unidirectional or duplex way, and the transitions between. Of course it
+can be the case that connections are no longer required in either direction,
+and such connections should be closed in an orderly manner. This must be a
+responsibility of the connection manager since it is the only component that
+can see both inbound and outbound sides to be able to see that a connection is
+no longer needed in either direction and hence not needed at all.
+
+In the next couple sections we will review what the inbound and outbound sides
+need to be able to do, and what service the connection manager needs to provide.
+
+\section{Outbound side: the P2P governor}
+
+A key component of the design for decentralisation is the P2P governor. It
+is responsible for:
+\begin{itemize}
+\item managing the selection of upstream peers;
+\item managing the transitions of upstream peers between cold/warm/hot states;
+\item continuously making progress towards the target number of peers in each
+      state; and
+\item adjusting these targets over time to achieve a degree of `churn'.
+      \todo[inline]{need cross reference.}
```

Comment:
This means we need a section on churn. Stealing from the most recent P2P blogpost might be a good idea